script.js

```diff
@@ -1,56 +1,57 @@
-/* Copyright end */
-/* eslint-enable */
+const commonmark = require('commonmark')
 
 if (document.body.children.length === 1 &&
 document.getElementsByTagName('pre')[0] !== undefined) {
   fetch(location.href).then(response => {
     if (response.headers.get('Content-Type') === 'text/markdown') {
       response.text().then(text => {
+        let parser = new commonmark.Parser()
+        let renderer = new commonmark.HtmlRenderer()
+
+        let parsed = parser.parse(text)
+        let rendered = renderer.render(parsed)
+
+        let walker = parsed.walker()
+        while(walker.current.type !== 'heading') walker.next()
```

Comment:
Should account for there being no headings in the document. Is there a `walker.hasNext`?

Reply:
Ah, you make a good point haha.
I'll refine the code.

Reply:
Fixed